tests/integration/logic.rs

```diff
@@ -0,0 +1,390 @@
+                    .into_conjunction(),
+                typeql_lang::parse_variable("$x has is-still-good true")
+                    .map_err(|err| typedb_client::Error::Other(format!("{err:?}")))?,
+            ).await?;
+            transaction.logic().put_rule(
+                "all-milk-is-good".to_string(),
+                typeql_lang::parse_pattern("{ $x isa milk; }")
+                    .map_err(|err| typedb_client::Error::Other(format!("{err:?}")))?
+                    .into_conjunction(),
+                typeql_lang::parse_variable("$x has is-still-good true")
+                    .map_err(|err| typedb_client::Error::Other(format!("{err:?}")))?,
+            ).await?;
+            transaction.commit().await?;
+        }
+
+        let session = Session::new(databases.get(common::TEST_DATABASE).await?, Data).await?;
+        let transaction = session.transaction(Write).await?;
+        let data = r#"insert $x isa milk, has age-in-days 5;"#;
+        let _ = transaction.query().insert(data)?;
+        let data = r#"insert $x isa milk, has age-in-days 10;"#;
+        let _ = transaction.query().insert(data)?;
+        let data = r#"insert $x isa milk, has age-in-days 15;"#;
+        let _ = transaction.query().insert(data)?;
+        transaction.commit().await?;
+
+        let with_inference_and_explanation = Options::new().infer(true).explain(true);
+        let transaction = session.transaction_with_options(Read, with_inference_and_explanation).await?;
+        let answer_stream = transaction.query().match_(
+            r#"match $x has is-still-good $a;"#,
+        )?;
+        let answers = answer_stream.try_collect::<Vec<ConceptMap>>().await?;
+
+        assert_eq!(3, answers.len());
+
+        assert!(answers.get(0).unwrap().explainables.is_some());
+        assert!(answers.get(1).unwrap().explainables.is_some());
+        assert!(answers.get(2).unwrap().explainables.is_some());
+
+        let age_in_days = transaction.concept().get_attribute_type(String::from("age-in-days")).await?.unwrap();
+        for ans in answers {
+            match ans.map.get("x").unwrap() {
+                Concept::Entity(entity) => {
+                    let attributes: Vec<Attribute> = entity.get_has(&transaction, vec![age_in_days.clone()], vec![])?.try_collect().await?;
+                    if attributes.first().unwrap().value == Value::Long(15) {
+                        assert_single_explainable_explanations(&ans, 1, 1, &transaction).await;
+                    } else {
+                        assert_single_explainable_explanations(&ans, 1, 2, &transaction).await;
+                    }
+                },
+                _ => panic!("Incorrect Concept type: {:?}", ans.map.get("x").unwrap()),
+            }
+        }
+
+        Ok(())
+    }
+}
+
+async fn assert_single_explainable_explanations(
+    ans: &ConceptMap,
+    explainables_count: usize,
+    explanations_count: usize,
+    transaction: &Transaction<'_>,
+) {
```

Comment:
If this asserts that we only have one explainable (which it seems that it does), why is `explainables_count` a parameter?